bittide/src/Bittide/ClockControl/DebugRegister.hs

```diff
@@ -0,0 +1,172 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Bittide.ClockControl.DebugRegister (
+  DebugRegisterData (..),
+  debugRegisterWb,
+) where
+
+import GHC.Records (HasField (..))
+
+import Clash.Prelude hiding (PeriodToCycles)
+
+import Protocols
+import Protocols.Wishbone
+
+import Data.Maybe (isJust)
+
+import Bittide.ClockControl (SpeedChange)
+import qualified Bittide.ClockControl.Callisto.Types as T
+import Bittide.Wishbone
+import Clash.Explicit.Signal.Extra (changepoints)
+import Clash.Functor.Extra
+import Clash.Sized.Vector.ToTuple (vecToTuple)
+
+data DebugRegisterData = DebugRegisterData
+  { reframingState :: T.ReframingState
+  , updatePeriod :: Unsigned 32
+  , updatePeriodMin :: Unsigned 32
+  , updatePeriodMax :: Unsigned 32
+  }
+
+instance
+  HasField
+    "reframingState"
+    (Signal dom DebugRegisterData)
+    (Signal dom T.ReframingState)
+  where
+  getField = fmap reframingState
+
+instance
+  HasField
+    "updatePeriod"
+    (Signal dom DebugRegisterData)
+    (Signal dom (Unsigned 32))
+  where
+  getField = fmap updatePeriod
+
+instance
+  HasField
+    "updatePeriodMin"
+    (Signal dom DebugRegisterData)
+    (Signal dom (Unsigned 32))
+  where
+  getField = fmap updatePeriodMin
+
+instance
+  HasField
+    "updatePeriodMax"
+    (Signal dom DebugRegisterData)
+    (Signal dom (Unsigned 32))
+  where
+  getField = fmap updatePeriodMax
+
+data ReframingStateKind = Detect | Wait | Done deriving (Generic, NFDataX, BitPack)
+
+{- | A Wishbone accessible debug register
+This interface holds values that the CPU should be reporting to the FPGA for debugging
+purposes. It also reports the update period coming from the CPU, as well as the
+minimum and maximum of the update period.
```

Comment:
I don't see that in here?

Reply:
Ah, it only _tracks_ it here, but does not expose it through its wishbone interface. Then why track it here?

Reply:
They are exposed through the `updatePeriod`, `updatePeriodMin`, and `updatePeriodMax` fields of the `DebugRegisterData` record type.